Cython/Compiler/PyrexTypes.py

```diff
@@ -3174,7 +3195,14 @@ def unique(somelist):
     return result
 
 def get_all_specialized_permutations(fused_types):
-    return _get_all_specialized_permutations(unique(fused_types))
+    fused_types = unique(fused_types)
+
+    # Filter out FusedType for which there is also a ConstFusedType
+    const_base_fused_types = [
+        t.base_fused_type for t in fused_types if isinstance(t, ConstFusedType)]
+    fused_types = [t for t in fused_types if t not in const_base_fused_types]
```

Comment:
Looks like this could be done by `unique()`, if the "type or base type" selection happens before calling it.

Reply:
I was expecting some side effects. This is no longer needed anyway.